fendermint/vm/interpreter/src/fvm/state/ipc.rs

```diff
@@ -202,6 +204,34 @@ impl<DB: Blockstore> GatewayCaller<DB> {
         encode_to_fvm_implicit(calldata.as_ref())
     }
 
+    pub fn decode_commit_parent_finality_return(
+        &self,
+        bytes: RawBytes,
+    ) -> anyhow::Result<(bool, IPCParentFinality)> {
+        let return_data = bytes
+            .deserialize::<BytesDe>()
+            .context("failed to deserialize return data")?;
+
+        let function = router::GATEWAYROUTERFACET_ABI
+            .functions
+            .get("commitParentFinality")
+            .ok_or_else(|| anyhow!("broken abi"))?
+            .get(0)
+            .ok_or_else(|| anyhow!("function not found, abi wrong?"))?;
+
+        if let Token::Tuple(mut tuple) = decode_function_data(function, return_data.0, false)? {
```

Comment:
`decode_function_data` is generic. I'm sure we can decode into the expected type instead of dealing with popping `Token`s.